stac_fastapi/extensions/stac_fastapi/extensions/core/collectionSearch/collectionSearch.py

```diff
@@ -0,0 +1,108 @@
+# encoding: utf-8
+"""Collection Search Extension."""
+from enum import Enum
+from typing import List, Type, Union
+
+import attr
+from fastapi import APIRouter, FastAPI
+from starlette.responses import JSONResponse, Response
+from stac_pydantic.api.collections import Collections
+
+from stac_fastapi.api.models import JSONSchemaResponse
+from stac_fastapi.api.routes import create_async_endpoint
+from stac_fastapi.types.core import AsyncCollectionSearchClient, CollectionSearchClient
+from stac_fastapi.types.extension import ApiExtension
+from stac_fastapi.types.search import BaseCollectionSearchGetRequest, BaseCollectionSearchPostRequest
+
+from .request import CollectionSearchExtensionGetRequest, CollectionSearchExtensionPostRequest
+
+class CollectionSearchConformanceClasses(str, Enum):
+    """Conformance classes for the Collection Search extension.
+
+    See
+    https://github.com/stac-api-extensions/collection-search
+    """
+
+    CORE = "https://api.stacspec.org/v1.0.0-rc.1/core"
+    COLLECTION_SEARCH = "https://api.stacspec.org/v1.0.0-rc.1/collection-search"
+    SIMPLE_QUERY = "http://www.opengis.net/spec/ogcapi-common-2/1.0/conf/simple-query"
+
+@attr.s
+class CollectionSearchExtension(ApiExtension):
+    """CollectionSearch Extension.
+
+    The collection search extension adds two endpoints which allow searching of 
+    collections via GET and POST:
+        GET /collection-search
+        POST /collection-search
```

Comment:
we should add a note that the extension seems to mention `GET /collections` and `POST /collections` but due to possible conflict with the `transaction` extension we should to use `/collection-search`

also wonder if we should use `/collections-search` or `/search/collections` 🤷‍♂️